File: C++/921 Minimum Add to Make Parentheses Valid.cpp
class Solution {
public:
    int minAddToMakeValid(string S) {
        stack<char> mystack;
        for (int i = 0; i < S.length(); i++) {
            char s = S[i];
            if (!mystack.empty() && s == ')' && mystack.top() == '(' ) mystack.pop();
            else mystack.push(s);
        }
        return mystack.size();
    }
};

File: Python/921 Minimum Add to Make Parentheses Valid.py
# Minimum Add to Make Parentheses Valid
class Solution(object):
    def minAddToMakeValid(self, S):
        """
        :type S: str
        :rtype: int
        """
        stack = []
        for ele in S:
            
            if ele == ')' and stack and stack[-1] == '(':
                stack.pop()
            else:
                stack.append(ele)
        return len(stack)